src/traceViewer/web/ui/filmStripView.tsx

```diff
@@ -0,0 +1,183 @@
+
+import { ContextEntry } from '../../traceModel';
+import './filmStripView.css';
+import { PageVideoTraceEvent } from '../../traceTypes';
+import { Boundaries } from '../components/geometry';
+import * as React from 'react';
+
+type MetaInfo = {
+  frames: number;
+  width: number;
+  height: number;
+  fps: number;
+  startTime: number;
+  endTime: number;
+};
+
+async function fetchMetaInfo(context: ContextEntry, video: PageVideoTraceEvent): Promise<MetaInfo | undefined> {
+  const response = await fetch(`context-artifact/${context.created.contextId}/${video.fileName}-metainfo.txt`);
+  const lines = (await response.text()).split('\n');
+  let framesLine = lines.find(l => l.startsWith('frame='));
+  if (!framesLine)
+    return;
+  framesLine = framesLine.substring(framesLine.lastIndexOf('frame='));
+  const framesMatch = framesLine.match(/frame=\s+(\d+)/);
+  const outputLineIndex = lines.findIndex(l => l.trim().startsWith('Output #0'));
+  const streamLine = lines.slice(outputLineIndex).find(l => l.trim().startsWith('Stream #0:0'))!;
+  const fpsMatch = streamLine.match(/, (\d+) fps,/);
+  const resolutionMatch = streamLine.match(/, (\d+)x(\d+)\D/);
+  const durationMatch = lines.find(l => l.trim().startsWith('Duration'))!.match(/Duration: (\d+):(\d\d):(\d\d.\d\d)/);
+  const duration = (((parseInt(durationMatch![1], 10) * 60) + parseInt(durationMatch![2], 10)) * 60 + parseFloat(durationMatch![3])) * 1000;
+  return {
+    frames: parseInt(framesMatch![1], 10),
+    width: parseInt(resolutionMatch![1], 10),
+    height: parseInt(resolutionMatch![2], 10),
+    fps: parseInt(fpsMatch![1], 10),
+    startTime: (video as any).timestamp,
+    endTime: (video as any).timestamp + duration
+  };
+}
+
+function imageURL(context: ContextEntry, fileName: string, index: number) {
+  const imageURLpadding = '0'.repeat(3 - String(index + 1).length);
+  return `context-artifact/${context.created.contextId}/${fileName}-${imageURLpadding}${index + 1}.png`;
+}
+
+export const FilmStrip: React.FunctionComponent<{
+  context: ContextEntry,
+  boundaries: Boundaries,
+  preview?: { time: number, clientX: number },
+}> = ({ context, boundaries, preview }) => {
+  const [metaInfos, setMetaInfos] = React.useState<Map<PageVideoTraceEvent, MetaInfo>>(new Map());
+  React.useEffect(() => {
+    async function initializeMetaInfo() {
+      const infos = new Map<PageVideoTraceEvent, MetaInfo>();
+      for (const page of context.pages) {
+        if (!page.video)
+          continue;
+        const metaInfo = await fetchMetaInfo(context, page.video);
+        if (metaInfo)
+          infos.set(page.video, metaInfo);
+      }
+      return infos;
+    }
+    setMetaInfos(new Map());
+    initializeMetaInfo().then(infos => setMetaInfos(infos));
+  }, [context]);
+
+  const ref = React.useRef<HTMLDivElement>(null);
+  const [measure, setMeasure] = React.useState(new DOMRect(0, 0, 10, 10));
+  React.useLayoutEffect(() => {
+    const target = ref.current;
+    if (!target)
```

Comment:
this is just for types right? The target should always be truthy because you always set the ref on the div.

Reply:
Yep.